packages/auth/amplify_auth_cognito/lib/src/credentials/cognito_keys.dart

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@internal
+library amplify_auth_cognito_dart.credentials.cognito_keys;
+
+import 'dart:collection';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:meta/meta.dart';
+
+/// Discrete keys stored for Cognito User Pool operations in secure storage.
+enum CognitoUserPoolKey {
+  /// The access token, serialized as a JWT.
+  accessToken,
+
+  /// The refresh token.
+  refreshToken,
+
+  /// The ID token, serialized as a JWT.
+  idToken,
+
+  /// The user ID.
+  userSub,
+
+  /// The device key.
+  deviceKey,
+
+  /// The device group key.
+  deviceGroupKey,
+}
+
+/// Discrete keys stored for Cognito Identity Pool operations in secure storage.
+enum CognitoIdentityPoolKey {
+  /// AWS Access Key ID
+  accessKeyId,
+
+  /// AWS Secret Access Key
+  secretAccessKey,
+
+  /// AWS Session Token
+  sessionToken,
+
+  /// AWS Credentials Expiration, encoded using ISO 8601.
+  expiration,
+
+  /// AWS Identity ID.
+  identityId,
+}
+
+/// Discrete keys stored for Hosted UI operations in secure storage.
+enum HostedUiKey {
+  /// The access token, serialized as a JWT.
+  accessToken,
+
+  /// The refresh token.
+  refreshToken,
+
+  /// The ID token, serialized as a JWT.
+  idToken,
+
+  /// The OAuth state string.
+  state,
+
+  /// The OAuth code verifier.
+  codeVerifier,
+
+  /// The OIDC nonce value.
+  nonce,
+}
+
+/// {@template amplify_auth_cognito_dart.cognito_identity_pool_keys}
```

Comment:
*nit:* Not sure if we want to update these too.

```suggestion
/// {@template amplify_auth_cognito.cognito_identity_pool_keys}
```